Library/Homebrew/cmd/missing.rb

```diff
@@ -6,15 +6,41 @@ module Homebrew
   def missing
     return unless HOMEBREW_CELLAR.exist?
 
-    ff = if ARGV.named.empty?
+    formulae = if ARGV.named.empty?
       Formula.installed
     else
       ARGV.resolved_formulae
     end
+    if ARGV.json == "v1"
+      missing = print_missing_json(formulae)
+    else
+      missing = print_missing(formulae)
+    end
+    Homebrew.failed = ARGV.resolved_formulae.any? && missing.any?
+  end
+
+  def print_missing(formulae)
+    verbose = ($stdout.tty? || ARGV.verbose?) && !ARGV.flag?("--quiet")
+    s = Set.new
+    missing = Diagnostic.missing_deps(formulae) do |name, m|
+      if verbose
+        print "#{name}: " if formulae.size > 1
+        puts "#{m.map(&:full_name) * " "}"
+      else
+        m.each { |f| s << f }
+      end
+    end
+    s.each { |f| puts f.full_name } unless verbose
+    missing
+  end
 
-    Diagnostic.missing_deps(ff) do |name, missing|
-      print "#{name}: " if ff.size > 1
-      puts "#{missing * " "}"
+  def print_missing_json(formulae)
+    json = []
+    missing = Diagnostic.missing_deps(formulae) do |name, m|
+      json << { :name => name,
+                :missing => m.collect(&:full_name) }
```

Comment:
A bit like in a comment further up, I think it would be more idiomatic Ruby (and easier to read), if `json` was created by doing something like the following (after invoking `Diagnostic.missing_deps` without a block).

```ruby
json = missing.map do |formula_name, missing_deps|
  { :name => formula_name, :missing => missing.deps.map(&:full_name) }
end
```